code/datums/components/crafting/crafting.dm

```diff
@@ -221,9 +237,12 @@
 	. = list()
 	var/data
 	var/amt
+	var/list/requirements = R.reqs + R.machinery
 	main_loop:
-		for(var/A in R.reqs)
-			amt = R.reqs[A]
+		for(var/A in requirements)
+			amt = R.reqs[A] || R.machinery[A]
```

Comment:
```suggestion
		for(var/path_key in requirements)
			amt = R.reqs[path] || R.machinery[path_key]
```
This implies the contents are type paths which work as keys in an associative list.